### PR TITLE
156 remastered dashboard

### LINKED ISSUE #156: Remastered dashboard
## Description
As a stakeholder I want to have a professional dashboard-page in order to give my customers a user experience

## Acceptance criteria
- [x] The design on the dashboard is on par with the Prodigo.se style.
- [x] A user should be directed to the dashboard when clicking on "prodigo portal"
- [x] Make headers and content more centered

### PR DESCRIPTION
**Pull request by Matte02**
## Summary

Visual upgrade to the dashboard. Now uses the same grid as the contract page, and scales properly for different screen sizes. Everything is more centered on the page as well. 

image
image
image
image

## Definition of done
- [X] All acceptance criteria are completed.
- [X] The code has (at least) 80% branch test coverage.
- [X] All tests pass.
- [ ] The code is reviewed and accepted by (at least) two other team members.
- [X] Views adhere to design style.

Closes #156

### CODE REVIEW

File: src/Client/Pages/Dashboard/DashboardPage.razor [hunk 1]
```diff
@@ -1,14 +1,21 @@
...
 
-<PageTitle>Prodigo Portal - Startsida</PageTitle>
-
-<div class="container">
-    <h2>Välkommen till Prodigo Portal!</h2>
-
-    <div class="d-flex justify-content-center w-100">
-        <StatusUpdates />
+<div class="dashboard-background container-fluid p-5 m-0">
+    <h2 class="text-center pb-3 title"> @WelcomeMessage</h2>
```
Comment (martinjonsson01) [inline src/Client/Pages/Dashboard/DashboardPage.razor:6]: nit:
```suggestion
    <h2 class="text-center pb-3 title">@WelcomeMessage</h2>
```

File: src/Client/Pages/Dashboard/DashboardPage.razor.css [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    padding: 20px;
+    margin: 10px;
+}
+
+.dashboard-background{
+    background-image: url("https://prodigo.se/wp-content/uploads/2021/05/cowomen-ZKHksse8tUU-unsplash-scaled.jpg");
+    padding: 0;
+    margin-left: 0;
+    margin-right: 0;
+    margin-top: 0;
```
Comment (martinjonsson01) [inline src/Client/Pages/Dashboard/DashboardPage.razor.css:13]: nit:
Why not set `margin: 0`? Doesn't seem like you're setting `margin-bottom` to anything.

File: src/Client/Pages/Dashboard/FavoriteCards.razor.css [hunk 3]
```diff
@@ -0,0 +1,9 @@
+﻿.favorite-grid {
+    display: flex;
+    flex-wrap: wrap;
+    justify-content: center;
+    gap: 25px 20px;
```
Comment (martinjonsson01) [inline src/Client/Pages/Dashboard/FavoriteCards.razor.css:5]: nit:
To allow for scaling in the future, it's better to use `rem` units.
```suggestion
    gap: 1.5625rem 1.25rem;
```

File: src/Client/Pages/Dashboard/DashboardPage.razor.css [hunk 4]
```diff
@@ -0,0 +1,22 @@
+﻿.dashboard-item-box{
+    background-color: var(--prodigo-light-blue);
+    border-radius: 20px;
+    padding: 20px;
+    margin: 10px;
```
Comment (martinjonsson01) [inline src/Client/Pages/Dashboard/DashboardPage.razor.css:5]: nit:
It's more scalable to use `rem` units instead.
```suggestion
    border-radius: 1.25rem;
    padding: 1.25rem;
    margin: 0.625rem;
```

**Review by martinjonsson01 — APPROVED** [review]
Splendid 👍

File: src/Client/Pages/Dashboard/DashboardPage.razor.css (hunk 4 above)
Comment (jonathan-lindqvist) [reply src/Client/Pages/Dashboard/DashboardPage.razor.css:5]: 🙏

File: src/Client/Pages/Dashboard/FavoriteCards.razor.css [hunk 5]
```diff
@@ -0,0 +1,9 @@
+﻿.favorite-grid {
+    display: flex;
+    flex-wrap: wrap;
+    justify-content: center;
```
Comment (jonathan-lindqvist) [inline src/Client/Pages/Dashboard/FavoriteCards.razor.css:4]: All of these are available in [Bootstrap](https://getbootstrap.com/docs/5.1/utilities/flex/), and can be added as classes directly.

File: src/Client/Pages/Dashboard/StatusUpdates.razor [hunk 6]
```diff
@@ -4,12 +4,13 @@
...
            Url="api/v1/status-updates"
            Context="statusUpdates">
-
-    <div class="list-group list-group-flush shadow w-50">
-        @foreach (StatusUpdate statusUpdate in statusUpdates.OrderByDescending(status => status.Timestamp))
-        {
-            <StatusUpdateItem Status="@statusUpdate"/>
-        }
+    <div>
+        <h3 class="text-center fw-bold"> Nyheter</h3>
```
Comment (jonathan-lindqvist) [inline src/Client/Pages/Dashboard/StatusUpdates.razor:8]:
```suggestion
        <h3 class="text-center fw-bold">Nyheter</h3>
```

File: src/Client/Shared/NavMenu.razor [hunk 7]
```diff
@@ -5,8 +5,11 @@
 
 <nav class="navbar navbar-expand-lg navbar-dark " style="background-color: #222939 ">
     <div class="container-fluid">
-        <a class="navbar-brand" href="">Prodigo Portal</a>
-        <button class="navbar-toggler" type="button"  data-bs-toggle="collapse" data-bs-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation">
+        <a class="navbar-brand"
+           href="@LogoUrl">
+            Prodigo Portal
+        </a>
```
Comment (jonathan-lindqvist) [inline src/Client/Shared/NavMenu.razor:11]: Nit: Autoformat?

File: src/Client/Shared/NavMenu.razor [hunk 8]
```diff
@@ -32,15 +35,19 @@
...
-
                     <li id="admin-nav-item" class="nav-item">
                         <NavLink class="nav-link" href="admin">
                             <span class="" aria-hidden="true"></span> Admin
                         </NavLink>
                     </li>
                 }
             </ul>
-            <Login />
+            <Login/>
```
Comment (jonathan-lindqvist) [inline src/Client/Shared/NavMenu.razor:45]: 🤖

**Review by jonathan-lindqvist — APPROVED** [review]
NEAT!

File: src/Client/Shared/NavMenu.razor (hunk 8 above)
Comment (martinjonsson01) [reply src/Client/Shared/NavMenu.razor:45]: 🤖

File: src/Client/Pages/Dashboard/StatusUpdates.razor [hunk 9]
```diff
@@ -4,12 +4,13 @@
 <FetchData TData="StatusUpdate[]"
            Url="api/v1/status-updates"
            Context="statusUpdates">
-
-    <div class="list-group list-group-flush shadow w-50">
-        @foreach (StatusUpdate statusUpdate in statusUpdates.OrderByDescending(status => status.Timestamp))
-        {
-            <StatusUpdateItem Status="@statusUpdate"/>
-        }
+    <div>
```
[the remark this answers is not in the document]
Comment (Matte02) [reply src/Client/Pages/Dashboard/StatusUpdates.razor:7]: image

File: src/Client/Pages/Dashboard/DashboardPage.razor.css (hunk 2 above)
Comment (Matte02) [reply src/Client/Pages/Dashboard/DashboardPage.razor.css:13]: Good catch!

File: src/Client/Pages/Dashboard/DashboardPage.razor.css [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+}
+
+.dashboard-background{
+    background-image: url("https://prodigo.se/wp-content/uploads/2021/05/cowomen-ZKHksse8tUU-unsplash-scaled.jpg");
```
[the remark this answers is not in the document]
Comment (Matte02) [reply src/Client/Pages/Dashboard/DashboardPage.razor.css:9]: How do I know if it's "super-high-resolution"?
As far as I can tell it's not that high resolution, 2560x1707.

File: src/Client/Pages/Dashboard/StatusUpdates.razor (hunk 9 above)
[the remark this answers is not in the document]
Comment (martinjonsson01) [reply src/Client/Pages/Dashboard/StatusUpdates.razor:7]: Huh, alrighty then

File: src/Client/Pages/Dashboard/DashboardPage.razor.css (hunk 10 above)
[the remark this answers is not in the document]
Comment (martinjonsson01) [reply src/Client/Pages/Dashboard/DashboardPage.razor.css:9]: Super high resolution would be something like 4000x3000, that is to say way above any normal screen resolution. But that size is probably fine 👍

**Comment by Matte02** [comment]
image
Added a message if there were no news. Still fine with the PR @martinjonsson01 and @jonathan-lindqvist?

**Comment by Matte02** [comment]
image
image

Updated status view for different screen sizes.

**Pull request merged**